syntax = "proto3";

package is;

import "google/protobuf/wrappers.proto";

// all units in meters
message AreaLimits {
  float xmin = 1;
  float xmax = 2;
  float ymin = 3;
  float ymax = 4;
}

// x and y in meters and length in pixels
message ReferentialProperties {
  float x = 1;
  float y = 2;
  uint32 length = 3;
}

enum RotateFlags {
  NONE = 0;
  ROTATE_90_CW = 1;
  ROTATE_180 = 2;
  ROTATE_90_CCW = 3;
}

message SkeletonsHeatmapOptions {
  string broker_uri = 1;
  string zipkin_host = 2;
  uint32 zipkin_port = 3;
  repeated uint32 group_ids = 4;
  AreaLimits limits = 5;
  float bins_step = 6;
  google.protobuf.FloatValue output_scale = 7;
  RotateFlags output_rotate = 15;
  int64 frame_id = 17;
  ReferentialProperties referential = 8;
  bool draw_grid = 9;
  bool flip_horizontal = 10;
  bool flip_vertical = 11;
  bool log_scale = 12;
  bool average_coordinates = 13;
  // if less or equal zero, the histogram will accumulate forever,
  // otherwise N latest samples will be stored to show heatmap of
  // last N samples. The maximum number of samples is 100000.
  int32 samples = 14;
  // Period in milliseconds of the rate that detections will be groupped
  // must be greater or equal than 200 and less or equal 1000 milliseconds
  int32 period_ms = 16;
}